File: java/akka-sample/service/src/main/scala/com/asiafrank/bangumi/service/mapper/ObjectMapperContextResolver.scala
package com.asiafrank.bangumi.service.mapper

import java.text.SimpleDateFormat
import javax.ws.rs.ext.{ContextResolver, Provider}

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.databind._
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule

/**
  * ObjectMapperContextResolver
  *
  * Created at 25/1/2017.
  *
  * @author asiafrank
  */
@Provider
class ObjectMapperContextResolver extends ContextResolver[ObjectMapper]{
  private val mapper = {
    val m = new ObjectMapper()
    m.registerModule(new JavaTimeModule)
      .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
      .configure(SerializationFeature.WRITE_ENUMS_USING_TO_STRING, true)
      .configure(DeserializationFeature.READ_ENUMS_USING_TO_STRING, true)
      .setSerializationInclusion(JsonInclude.Include.NON_EMPTY)
      .setDateFormat(new SimpleDateFormat("yyyy-MM-dd HH:mm a z"))
    m
  }

  override def getContext(t: Class[_]): ObjectMapper = mapper
}

File: java/jersey/src/main/java/com/asiafrank/learn/jersey/mapper/ObjectMapperContextResolver.java
package com.asiafrank.learn.jersey.mapper;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;

import javax.ws.rs.ext.ContextResolver;
import javax.ws.rs.ext.Provider;
import java.text.SimpleDateFormat;

/**
 * ObjectMapperContextResolver
 * <p>
 * </p>
 * Created at 2/8/2017.
 *
 * @author zhangxf
 */
@Provider
public final class ObjectMapperContextResolver implements ContextResolver<ObjectMapper> {
    private final ObjectMapper mapper = getMapper();

    @Override
    public ObjectMapper getContext(Class<?> type) {
        return mapper;
    }

    private ObjectMapper getMapper() {
        ObjectMapper m = new ObjectMapper();
        m.registerModule(new JavaTimeModule())
                .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
                .configure(SerializationFeature.WRITE_ENUMS_USING_TO_STRING, true)
                .configure(DeserializationFeature.READ_ENUMS_USING_TO_STRING, true)
                .setSerializationInclusion(JsonInclude.Include.NON_EMPTY)
                .setDateFormat(new SimpleDateFormat("yyyy-MM-dd HH:mm a z"));
        return m;
    }
}
